src/parse/grammar/gast/gast_public.ts

```diff
@@ -248,6 +278,8 @@ export function serializeProduction(node: IProduction): ISerializedGast {
     } else if (node instanceof RepetitionWithSeparator) {
         return <ISerializedTerminalWithSeparator>{
             type: "RepetitionWithSeparator",
+            name: node.name,
+            idx: node.idx,
```

Comment:
👍

Reply:
Btw, `name` can be undefined on a lot of `IProduction`s if you don’t pass it in via the options in the constructor. Setting it to an empty string initially causes errors to be thrown. I had to add `name: undefined` to get some tests to pass and make sure we don’t define name when the name in options is set but undefined.

Reply:
The name is optional, so setting it to empty string does not make sense...

When you say it is undefined do you mean the property exists but is undefined 
or not property exists at all?

Reply:
The property exists and is undefined on certain productions (any production that’s tested here where I added a name) `test/parse/grammar/gast_spec.ts`.
I just found it to be suprising behavior b/c I thought `name: string` implied a property is defined in typescript but apparently that's only when you run typescript with strictNullChecks.